play/src/components/ScoreIndicator.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState, useEffect } from 'react';
+import { makeStyles, ThemeProvider} from '@mui/styles';
+import { Typography } from "@mui/material";
+import { isNullOrUndefined } from '@righton/networking';
+
+interface ScoreIndicatorProps {
+  newPoints: number;
+  score: number | null;
+}
+
+export default function ScoreIndicator({
+  newPoints,
+  score,
+}:ScoreIndicatorProps) {
+  const classes = useStyles();
+  const [currentScore, setCurrentScore] = useState(isNullOrUndefined(score) ? 0 : score);
+
+  // adds an eventLister to add the new points to the existing score when the animation completes
+  useEffect(() => {
+    const element = document.getElementById('newPointsAnimation');
+    const handleAnimationEnd = () => {
+      setCurrentScore(prevScore => (prevScore+newPoints));
+    }
+    element?.addEventListener('animationend', handleAnimationEnd);
+    return () => {
+      element?.removeEventListener('animationend', handleAnimationEnd);
+    }
+  });
+
+  return(
+    <>
+      <div id='newPointsAnimation' className={classes.newPointsAnimation}>
+        { (newPoints && newPoints > 0) ? 
+          <div className={classes.newPointsContainer}>
+            <Typography className={classes.text}> {`+${newPoints}`} </Typography>
+          </div>
+        : 
+          null
+        }
+      </div>
+      <div className={classes.scoreContainer}> 
+        <Typography className={classes.text}> {isNullOrUndefined(currentScore) ? 0 : currentScore} </Typography>
```

Comment:
Intentional spaces in the beginning and end?

Reply:
removed